### DISCUSSION TITLE
patterns/singleton/

### DISCUSSION THREAD

**Discussion by giscus** [iluwatar/java-design-patterns-vuepress-web#77] [Announcements]
# patterns/singleton/

Intent Ensure a class only has one instance, and provide a global point of access to it. Explanation Real-world example There can only be one ivory tower where the wizards study their magic. The same enchanted ivory tower is always used by the wizards. The ivory tower here is a singleton.

https://java-design-patterns.com/patterns/singleton/

**Comment by jojijoy15** [comment]
What is the difference in implementation of ```BillPughImplementation``` and ```InitializingOnDemandHolderIdiom```. I dont see much difference, seems to use the same principle, as they depend on gurantees offered by Java language.
